### ISSUE TITLE
ArgumentList bug: whitespace characters are being treated as a argument.

### ISSUE THREAD

**Issue by czyjerry** [WeAreROLI/JUCE#471]
I've pulled the lastest commits from the `develop` branch.
And the minimal reproduction code is shown below:

```
ArgumentList mylist("test", "abc def ");
std::cout << mylist.size() << std::endl;
```

This will ouput "3", but it's expected to output "2", as whitespace shouldn't be treated as a argument.
I also tested:
" abc def", 
"abc def	" (tab character),
"	abc def" (tab character),
"abc  def" (two space), 
"abc		def"(two tabs)

These all output "3" instead of the expected "2".
And strings with more spaces like "abc   def", "abc def   ", will output even larger numbers.
So I think this is a bug of the "ArgumentList".

**Comment by czyjerry** [comment]
Here is the link to the corresponding forum topic: https://forum.juce.com/t/argumentlist-bug-whitespace-characters-are-being-treated-as-a-argument/31397

**Comment by ed95** [comment]
Fixed in [38fb341](https://github.com/WeAreROLI/JUCE/commit/38fb341ee402c6ec67cab157414eb76de649eb07), thanks for reporting.
